test/org/sagebionetworks/bridge/BridgeUtilsTest.java

```diff
@@ -387,6 +387,22 @@ public void getLocalDateWithError() {
         BridgeUtils.getLocalDateOrDefault("2017-05-10T05:05:10.000Z", null);
     }
     
+    @Test
+    public void toSynapseFriendlyName() {
+        assertEquals("This is a .-_ synapse Friendly Name3",
+                BridgeUtils.toSynapseFriendlyName("This (is a).-_ synapse Friendly Name3 "));
+    }
+    
+    @Test
+    public void nullToSynapseFriendlyName() {
+        assertEquals("", BridgeUtils.toSynapseFriendlyName(null));
+    }
+    
+    @Test
+    public void emptyStringToSynapseFriendlyName() {
+        assertEquals("", BridgeUtils.toSynapseFriendlyName("  #"));
+    }
```

Comment:
If the result is an empty string, should that be an error?

Reply:
Now throws an IllegalArgumentException